diff --git a/CMakeLists.txt b/CMakeLists.txt
index 129b1e3..e631539 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,60 +2,36 @@ cmake_minimum_required(VERSION 3.0.0)
 project(glsl-language-server)
 
 find_package(Threads REQUIRED)
-
-add_subdirectory(externals/glslang EXCLUDE_FROM_ALL)
-include_directories(
-    externals/glslang/
-    externals/glslang/glslang/Public
-    externals/glslang/glslang/MachineIndependent
-    externals/glslang/StandAlone
-)
-
-add_subdirectory(externals/fmt EXCLUDE_FROM_ALL)
-include_directories(
-    externals/fmt
-)
-
-add_subdirectory(externals/json EXCLUDE_FROM_ALL)
-include_directories(
-    externals/json/include
-)
-
+find_package(SPIRV-Tools-opt REQUIRED)
+find_package(glslang REQUIRED)
+find_package(fmt REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(CLI11 REQUIRED)
 add_library(mongoose externals/mongoose/mongoose.c)
-include_directories(
-    externals/mongoose
-)
-
-include_directories(
-    externals/CLI11/include)
+include_directories(externals/mongoose)
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 
-file(GLOB SOURCES
-    src/*.cpp
-)
+file(GLOB SOURCES src/*.cpp)
 
 include_directories(src)
-add_executable(glslls
-    ${SOURCES}
-    externals/glslang/StandAlone/ResourceLimits.cpp
-)
+add_executable(glslls ${SOURCES})
 
-if (CMAKE_SYSTEM_NAME MATCHES Darwin)
-    set(stdfs)
+if(CMAKE_SYSTEM_NAME MATCHES Darwin OR CMAKE_SYSTEM_NAME MATCHES FreeBSD)
+  set(stdfs)
 else()
-    set(stdfs stdc++fs)
+  set(stdfs stdc++fs)
 endif()
 
-target_link_libraries(glslls
-    ${CMAKE_THREAD_LIBS_INIT}
-    glslang
-    nlohmann_json
-    mongoose
-    ${stdfs}
-    SPIRV
-    fmt::fmt-header-only
-)
-
-install(TARGETS glslls
-        RUNTIME DESTINATION bin)
+target_link_libraries(
+  glslls
+  ${CMAKE_THREAD_LIBS_INIT}
+  glslang::glslang
+  nlohmann_json::nlohmann_json
+  mongoose
+  ${stdfs}
+  glslang::SPIRV
+  glslang::glslang-default-resource-limits
+  fmt::fmt-header-only)
+
+install(TARGETS glslls RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
diff --git a/src/includer.hpp b/src/includer.hpp
index 5521eb6..a321474 100644
--- a/src/includer.hpp
+++ b/src/includer.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include "ShaderLang.h"
+#include <glslang/Public/ShaderLang.h>
 #include "workspace.hpp"
 
 class FileIncluder : public glslang::TShader::Includer {
diff --git a/src/main.cpp b/src/main.cpp
index 05fe10c..e10b6f5 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1,15 +1,15 @@
-#include "CLI/CLI.hpp"
+#include <CLI/CLI.hpp>
 
-#include "fmt/format.h"
-#include "fmt/ostream.h"
+#include <fmt/format.h>
+#include <fmt/ostream.h>
 
-#include "nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 
 #include "mongoose.h"
 
-#include "ResourceLimits.h"
-#include "ShaderLang.h"
-#include "Initialize.h"
+#include <glslang/Public/ResourceLimits.h>
+#include <glslang/Public/ShaderLang.h>
+#include <glslang/MachineIndependent/Initialize.h>
 
 #include <cstdint>
 #include <filesystem>
@@ -571,14 +571,14 @@ int main(int argc, char* argv[])
     app.add_option("-l,--log", logfile, "Log file");
     app.add_option("--debug-symbols", symbols_path, "Print the list of symbols for the given file");
     app.add_option("--debug-diagnostic", diagnostic_path, "Debug diagnostic output for the given file");
-    app.add_option("-p,--port", port, "Port", true)->excludes(stdin_option);
+    app.add_option("-p,--port", port, "Port")->excludes(stdin_option);
     app.add_option("--target-env", client_api,
             "Target client environment.\n"
-            "    [vulkan vulkan1.0 vulkan1.1 vulkan1.2 vulkan1.3 opengl opengl4.5]", true);
+            "    [vulkan vulkan1.0 vulkan1.1 vulkan1.2 vulkan1.3 opengl opengl4.5]");
     app.add_option("--target-spv", spirv_version,
             "The SPIR-V version to target.\n"
             "Defaults to the highest possible for the target environment.\n"
-            "    [spv1.0 spv1.1 spv1.2 spv1.3 spv1.4 spv1.5 spv1.6]", true);
+            "    [spv1.0 spv1.1 spv1.2 spv1.3 spv1.4 spv1.5 spv1.6]");
 
     try {
         app.parse(argc, argv);
diff --git a/src/messagebuffer.hpp b/src/messagebuffer.hpp
index 423056d..0394fd9 100644
--- a/src/messagebuffer.hpp
+++ b/src/messagebuffer.hpp
@@ -1,7 +1,7 @@
 #ifndef MESSAGEBUFFER_H
 #define MESSAGEBUFFER_H
 
-#include "nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 
 #include <string>
 #include <tuple>
